kube-client/src/client/auth/mod.rs

```diff
@@ -474,22 +475,32 @@ fn token_from_gcp_provider(provider: &AuthProviderConfig) -> Result<ProviderToke
     }
 }
 
-fn extract_value(json: &serde_json::Value, path: &str) -> Result<String, Error> {
-    let pure_path = path.trim_matches(|c| c == '"' || c == '{' || c == '}');
-    match jsonpath_select(json, &format!("${pure_path}")) {
-        Ok(v) if !v.is_empty() => {
-            if let serde_json::Value::String(res) = v[0] {
-                Ok(res.clone())
-            } else {
-                Err(Error::AuthExec(format!(
-                    "Target value at {pure_path:} is not a string"
-                )))
-            }
-        }
+fn extract_value(json: &serde_json::Value, context: &str, path: &str) -> Result<String, Error> {
+    let parsed_path = {
+        let trimmed = path.trim_matches(|c| c == '"' || c == '{' || c == '}');
+        JsonPathInst::from_str(trimmed).map_err(|err| {
+            Error::AuthExec(format!(
+                "Failed to parse {context:?} as a JsonPath: {path}\n
+                 Error: {err}"
+            ))
+        })?
+    };
 
-        Err(e) => Err(Error::AuthExec(format!("Could not extract JSON value: {e:}"))),
+    let res = parsed_path.find_slice(json);
 
-        _ => Err(Error::AuthExec(format!("Target value {pure_path:} not found"))),
+    let Some(res) = res.into_iter().next() else {
+        return Err(Error::AuthExec(format!(
+            "Target {context:?} value {path:?} not found"
+        )));
+    };
+
+    if let Some(val) = res.as_str() {
+        Ok(val.to_owned())
+    } else {
+        Err(Error::AuthExec(format!(
+            "Target {:?} value {:?} is not a string: {:?}",
+            context, path, *res
```

Comment:
for cleanliness sake, may want to format this string the same way as the other two above: `{context:?}`, `{path:?}`, `{*res:?}`

Reply:
`{*res:?}` is invalid.
Inline names can not contain expressions, only symbols, optionally path qualified.

As soon as a single name is not inlined, it is clearer not to inline any of them.  Mix of inlined and non-lined names can be hard to follow.

Reply:
oh ya ok did not know that. good to know. thank you for the clarifications!